runtime/runtime/src/verifier.rs

```diff
@@ -295,17 +295,7 @@ pub(crate) fn validate_receipt(
         }
     }
 
-    // We retain these checks here as to maintain backwards compatibility
-    // with AccountId validation since we illegally parse an AccountId
-    // in near-vm-logic/logic.rs#fn(VMLogic::read_and_parse_account_id)
-    AccountId::validate(receipt.predecessor_id().as_ref()).map_err(|_| {
-        ReceiptValidationError::InvalidPredecessorId {
-            account_id: receipt.predecessor_id().to_string(),
-        }
-    })?;
-    AccountId::validate(receipt.receiver_id().as_ref()).map_err(|_| {
-        ReceiptValidationError::InvalidReceiverId { account_id: receipt.receiver_id().to_string() }
-    })?;
+    // Account ID validation is now handled in read_and_parse_account_id
```

Comment:
please remove added comments in this file, they are not really useful in the future when we no longer concerned with how `read_and_parse_account_id` used to be implemented